tests/association/test_positioning.py

```diff
@@ -306,3 +316,24 @@ def test_distance_between_points(input, expected):
 def test_manhatan_distance_between_points(input, expected):
     distance = Positioning.manhattan_distance_between_points(input[0], input[1])
     assert round(distance, 2) == expected
+
+def test_tokens_within_bounds(bbox_token_page):
+    box = generate_mapped_pred(1260, 1465, 610, 1210, page_num=0)
+    positioning = Positioning()
+    bounds = positioning.tokens_within_bounds(box, bbox_token_page)
+    for token in bounds:
+        assert "false" not in token["text"] and "edge" not in token["text"]
```

Comment:
all of these should be separate tests-- generally you want to start with a simple test, where you can just have one token, and say `test_correctly_identifies_token_within_bbox` and then the inverse `test_correctly_excludes_token_outside`, etc. etc. then have a test for your partial overlaps

Reply:
this way you can actually include one dummy token inside and it is very clean on observation that it is correctly including/excluding-- you'll also obviously want a test to ensure that it can handle multiple values (could do two for example in one test)

Reply:
Broke tests out into multiples to add more granularity.